stored_procedures/update_run_state.sql

```diff
@@ -0,0 +1,30 @@
+-- drop the stored procedure
+DROP PROCEDURE IF EXISTS `biosero_uat`.`updateRunState`;
+
+-- create the stored procedure
+DELIMITER $$
+
+-- Updates a destination plate well row with a linked control well
+-- Q. Do we want to pass in end_time as a parameter?
+CREATE PROCEDURE `biosero_uat`.`updateRunState` (
+  IN input_automation_system_name VARCHAR(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci,
+  IN input_system_run_id INT,
+  IN input_state VARCHAR(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci
+)
+BEGIN
+
+  UPDATE `biosero_uat`.`automation_system_runs`
+  SET
+    end_time = now(),
```

Comment:
Is it always the case on updating the run state that the end time should be added? Thinking only if a state e.g. 'paused' existed in the future that the end_time wouldn't want to be updated

Reply:
state is an ENUM, so I've changed it to SET based on a CASE statement based on input state. if it's completed or aborted it sets the end time, else leaves it. Our ENUM doesn't currently contain a paused state, but maybe it should (would they send started state again when they continued?